### ISSUE TITLE
flattenScope fails because importMap is empty

### ISSUE THREAD

**Issue by KoltesDigital** [jspm/jspm-cli#2428]
Hi,

I'm following the guide to make a new project with TypeScript and Aurelia. I was unable to generate the import map because the DEW put a static reference to aurelia-loader-nodejs. But that's for another ticket... I reproduced the essentials of this project, and it already fails to generate the import map.

Using https://github.com/KoltesDigital/example-ts-jspm-aurelia/commit/4156461895bba02719f132ca68306b19a4453541:

```
> jspm run compile
> jspm map ./dist/app.js -o importmap.json --flat-scope --map-base .
err  TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at Object.flattenScopes (C:\Projects\example-jspm-aurelia\node_modules\jspm\lib\map\utils.js:117:32)
    at Object.filterMap (C:\Projects\example-jspm-aurelia\node_modules\jspm\lib\map\index.js:331:17)
```

Inspecting `flattenScope`, it appears the function is called with an empty object as parameter.

BTW, importing other files requires `.js` at the end, even on the TS side. Could something avoid that?

Thanks!

**Comment by guybedford** [comment]
Thanks for trying it out and reporting the bug. The issue here is that if there are no scopes to flatten, it's not checking that case. I've added a fix in https://github.com/jspm/jspm-cli/commit/03ad7036c7954265baa67e61a07d5205d3d8b3e5.

In the mean time, until the release, you can simply omit the `--flat-scope` flag in cases where it throws this error (that is, if it throws, you don't need it!).

As for extensions - this is not something that will be supported in import maps in browser so explicit extensions are necessary at this point, yes.

**Comment by KoltesDigital** [comment]
Thanks for the reactivity! Indeed it works without the flag. New issue soon :)

**Comment by guybedford** [comment]
Released in 2.0.0-beta.1.
